components/core/src/clp/clp/FileDecompressor.cpp

```diff
@@ -76,4 +118,125 @@ bool FileDecompressor::decompress_file(
+        if (ErrorCode_Success != error_code) {
+            SPDLOG_ERROR(
+                    "Failed to create directory structure {}, errno={}",
+                    output_dir.c_str(),
+                    errno
+            );
+            return false;
+        }
+    }
+
+    boost::filesystem::path temp_ir_path{temp_output_dir};
+    temp_ir_path /= m_encoded_file.get_id_as_string() + ".temp." + ir::get_ir_extension_name();
+
+    auto const& file_orig_id = m_encoded_file.get_orig_file_id_as_string();
+    auto begin_message_ix = m_encoded_file.get_begin_message_ix();
+
+    LogEventSerializer<four_byte_encoded_variable_t> ir_serializer;
+    // Open output IR file
+    error_code = ir_serializer.open(temp_ir_path.string(), m_encoded_file.get_begin_ts());
+    if (ErrorCode_Success != error_code) {
+        return false;
+    }
+
+    while (archive_reader.get_next_message(m_encoded_file, m_encoded_message)) {
+        if (!archive_reader
+                     .decompress_message_without_ts(m_encoded_message, m_decompressed_message))
+        {
+            SPDLOG_ERROR("Failed to decompress message");
+            return false;
+        }
+        auto const message_size_as_ir = ir::get_approximated_ir_size(
+                m_decompressed_message,
+                m_encoded_message.get_vars().size()
+        );
+        if (message_size_as_ir + ir_serializer.get_serialized_size() > ir_target_size) {
+            ir_serializer.close();
+
+            auto end_message_ix = begin_message_ix + ir_serializer.get_num_log_events();
+            if (false
+                == rename_and_move_ir(
+                        temp_ir_path,
+                        output_dir,
+                        file_orig_id,
+                        begin_message_ix,
+                        end_message_ix
+                ))
+            {
+                return false;
+            }
+            begin_message_ix = end_message_ix;
+
+            error_code = ir_serializer.open(
+                    temp_ir_path.string(),
+                    m_encoded_message.get_ts_in_milli()
+            );
+            if (ErrorCode_Success != error_code) {
+                return false;
+            }
+        }
+
+        error_code = ir_serializer.serialize_log_event(
+                m_decompressed_message,
+                m_encoded_message.get_ts_in_milli()
+        );
+        if (ErrorCode_Success != error_code) {
+            SPDLOG_ERROR("Failed to serialize log event: {}", m_decompressed_message.c_str());
+            return false;
+        }
+    }
+    auto end_message_ix = begin_message_ix + ir_serializer.get_num_log_events();
+    ir_serializer.close();
+
+    if (false
+        == rename_and_move_ir(
+                temp_ir_path,
+                output_dir,
+                file_orig_id,
+                begin_message_ix,
+                end_message_ix
+        ))
+    {
+        return false;
+    }
+
```

Comment:
I can tell the reason why you don't remove `temp_output_dir` (since in the current implementation you don't know if it already exists before this call), but can we add a comment to explain this behavior? It is weird to create some temp resources without cleaning up.